File: ruby-sdk/fyipe/lib/oneuptime/oneuptimeListener.rb
require_relative 'timelineManager'
require_relative 'util'

class OneUptimeListener
    def initialize(eventId, options)
        # start the timeline manager
        @timelineObj = OneUptimeTimelineManager.new(options)
        @currentEventId = eventId
        @utilObj = Util.new(options)
    end

    def logErrorEvent(content, category = 'exception')
        timelineObj = {}
        timelineObj["category"]= category
        timelineObj["data"]= content
        timelineObj["type"]= @utilObj.getErrorType('ERROR')
        timelineObj["eventId"]= @currentEventId

        # add timeline to the stack
        @timelineObj.addToTimeline(timelineObj)
    end

    def logCustomTimelineEvent(timelineObj)
        timelineObj["eventId"] = @currentEventId

        # add timeline to the stack
        @timelineObj.addToTimeline(timelineObj)
    end

    def getTimeline()
        # this always get the current state of the timeline array
        return @timelineObj.getTimeline()
    end

    def clearTimeline(eventId)
        # set a new eventId
        @currentEventId = eventId
        # this will reset the state of the timeline array
        return @timelineObj.clearTimeline()
    end
    
end

File: Python-sdk/oneuptime_sdk/oneuptime_sdk/oneuptimeListener.py
from .timelineMagager import OneUptimeTimelineManager
from .util import Util


class OneUptimeListener:
    def __init__(self, eventId, options):
        # start the timeline manager
        self.timelineObj = OneUptimeTimelineManager(options)
        self.currentEventId = eventId;
        self.utilObj = Util(options)
    
    def logErrorEvent(self, content, category = 'exception'):

        timelineObj = {
            "category": category,
            "data": content,
            "type": self.utilObj.getErrorType().ERROR,
            "eventId": self.currentEventId
        } 

        # add timeline to the stack
        self.timelineObj.addToTimeline(timelineObj)

    def logCustomTimelineEvent(self, timelineObj):
        timelineObj["eventId"] = self.currentEventId

        # add timeline to the stack
        self.timelineObj.addToTimeline(timelineObj)
    
    def getTimeline(self):
        # this always get the current state of the timeline array
        return self.timelineObj.getTimeline()
    
    def clearTimeline(self, eventId):
        # set a new eventId
        self.currentEventId = eventId
        # this will reset the state of the timeline array
        return self.timelineObj.clearTimeline()
    
    
    